qualtran/bloqs/comparison_gates.ipynb

```diff
@@ -0,0 +1,957 @@
+    "    yield cirq.CNOT(anc[0], anc[1])  # OR if p < q.\n",
+    "\n",
+    "    yield cirq.CNOT(anc[0], less_than)\n",
+    "    yield cirq.CNOT(q, z)\n",
+    "    yield cirq.CNOT(anc[1], greater_than)\n",
+    "\n",
+    "    # Undo changes done the data and ancilla qubits.\n",
+    "    yield cirq.CNOT(anc[0], anc[1])\n",
+    "    yield cirq.CNOT(q, anc[1])\n",
+    "    yield cirq.X(anc[1])\n",
+    "    yield cirq.X(q)\n",
+    "    yield cirq.CNOT(p, q)\n",
+    "    yield cirq.X(p)\n",
+    "    yield cirq.CCNOT(p, q, anc[0])\n",
+    "    yield cirq.X(p)\n",
+    "\n",
+    "\n",
+    "less_than = cirq.NamedQubit('less_than')\n",
+    "greater_than = cirq.NamedQubit('greater_than')\n",
+    "single_qubit_compare_circuit = cirq.Circuit(\n",
+    "    single_qubit_compare(\n",
+    "        first_quantum_number[0], second_quantum_number[0], z, less_than, greater_than\n",
+    "    )\n",
+    ")\n",
+    "single_qubit_compare_circuit"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Now we check individual possiblities.\n",
+    "# The output for this and following circuits\n",
+    "# will be three qubits in the order [less_than, equal, greater_than]\n",
+    "check_each_possibility(\n",
+    "    single_qubit_compare_circuit,\n",
+    "    [first_quantum_number[0], second_quantum_number[0]],\n",
+    "    [less_than, z, greater_than],\n",
+    "    quantum_classical=False,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "check_uniform_superposition(\n",
+    "    single_qubit_compare_circuit,\n",
+    "    [first_quantum_number[0], second_quantum_number[0]],\n",
+    "    [less_than, z, greater_than],\n",
+    "    quantum_classical=False,\n",
+    ")"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "A slightly different circuit is given in the paper which we implemented in `cirq_ft` as `cirq_ft.algos.SingleQubitCompare` and has a cost of only one CCNOT (i.e. Toffoli = 4T).\n",
+    "\n",
+    "`cirq_ft.algos.SingleQubitCompare` is implemented using measurement based uncomputation so that its adjoint has T count of zero."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### $n=2$ Case\n",
+    "In this case the two qubit numbers $P$ and $Q$ can be written as\n",
+    "$$\n",
+    "P = 2*p_1 + p_0\\\\\n",
+    "Q = 2*q_1 + q_0\n",
+    "$$\n",
+    "\n",
+    "We will do this in two steps. First we update qubits such that $\\textit{sign}(P - Q) = \\textit{sign}(p_0^f - q_0^f)$ where $p_0^f$ and $q_0^f$ are the final values of $p_0$ and $q_0$ respectively. This way the comparison result between the two numbers can be extracted using `cirq_ft.algos.SingleQubitCompare` applied on $p_0^f$ and $q_0^f$.\n",
```

Comment:
`\textit{sign}`? a) won't this already be italics b) we usually write these operators with `\mathrm{}` to make them _not_ italic